server/sonar-server/src/main/java/org/sonar/server/rule/ws/ListAction.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.rule.ws;
+
+import com.google.common.base.Strings;
+import org.apache.ibatis.session.ResultContext;
+import org.apache.ibatis.session.ResultHandler;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.rule.RuleDto;
+import org.sonar.server.plugins.MimeTypes;
+import org.sonarqube.ws.Rules.ListResponse;
+
+public class ListAction implements RulesWsAction {
+
+  private final DbClient dbClient;
+
+  public ListAction(DbClient dbClient) {
+    this.dbClient = dbClient;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    controller
+      .createAction("list")
+      .setDescription("List of rules, excluding the manual rules and the rules with status REMOVED. JSON format is not supported for response.")
+      .setSince("5.2")
+      .setInternal(true)
+      .setHandler(this);
+  }
+
+  @Override
+  public void handle(Request wsRequest, Response wsResponse) throws Exception {
+    DbSession dbSession = dbClient.openSession(false);
+    final ListResponse.Builder listResponseBuilder = ListResponse.newBuilder();
+    final ListResponse.Rule.Builder ruleBuilder = ListResponse.Rule.newBuilder();
+    try {
+      dbClient.ruleDao().selectEnabledAndNonManual(dbSession, new ResultHandler() {
```

Comment:
extract in a private static class with listReponseBuilder and ruleBuilder as parameter